scalafmt-tests/src/test/resources/scala3/OptionalBraces.stat

```diff
@@ -4277,17 +4277,14 @@ object a:
      )(b: B,
        bs: B*
      )(implicit
-       implicit
-       c: C) {}
+       implicit c: C) {}
```

Comment:
Actually, how does it parse? Two implicits in a row doesn't parse for me :thinking:

Reply:
well, then we should change our scalameta parser :) or the grammar:

under https://dotty.epfl.ch/docs/internals/syntax.html#type-and-value-parameters
```
ClsParamClauses   ::=  {ClsParamClause} [[nl] ‘(’ [‘implicit’] ClsParams ‘)’]
...
ClsParam          ::=  ...
                       [{Modifier} (‘val’ | ‘var’) | ‘inline’] Param
```
and under https://dotty.epfl.ch/docs/internals/syntax.html#bindings-and-imports
```
Modifier          ::=  LocalModifier
...
LocalModifier     ::=  ...
                    |  ‘implicit’
```
so nothing really prevents multiple implicits. I don't know if `{...}` means multiple _different_ ones, but even in that case the first parameter is definitely covered.

Reply:
I think the grammar then. Both the compiler and scalameta fail to parse it.

Reply:
wait, if scalameta did fail to parse it, then scalafmt wouldn't be able to format it. perhaps a different dialect?

Reply:
in any case, this specific change doesn't really touch the multiple implicits logic, it's just _some_ of the tests which are affected. unless, of course, the compiler doesn't allow explicit implicits all of a sudden, like:
```
...(
  implicit a: A,
  implicit b: B
)
```

Reply:
Ach wait, that's my bad. It is possible for classes and not for defs, which I didn't realize. But it fails for both classes and defs when trying to compiler, so this seems like an issue in scalameta parser only.

But yeah, it's not strictly related to this PR I was just confused how this even parsed.

Reply:
should we open an issue in the parser and describe how it works? does it allow explicit implicits, does it require unique modifiers per parameter, does it only check consecutive ones? are there modifiers which aren't allowed at all?

ideally, they'd revise the grammar...

Reply:
I think this is an omission really and we can open an issue in the parser, but most other modifiers should be handled properly
just implicit/using are a weird case

Reply:
@tgodzik could you please check this:
```
class A(implicit implicit val a: String):
  // foo
```
in my Intellij IDE it works (and without `val` doesn't).

Reply:
There is an error from the compiler: `Repeated modifier implicit`